[YUNIKORN-1031] Add Chinese translation for homepage

cdmikechen opened the pull request:
At present, most website pages and some design documents have been completed.
There's a lot of work to do later. It is expected to improve the subsequent parts before version 1.0.

The translation parts mainly come from my personal repositories: <https://github.com/cdmikechen/yunikorn-site-zh>

yuchaoran2011 commented on i18n/zh-cn/docusaurus-plugin-content-docs/version-0.12.1/api/cluster.md:
```diff
@@ -0,0 +1,128 @@
+
+## 集群
+
+返回由YuniKorn调度程序管理的群集常规信息，信息内容包括应用程序和容器的数量（包括总计、失败、挂起、正在运行和已完成的数量）。
```
```suggestion
返回由YuniKorn调度程序管理的集群常规信息。信息内容包括应用程序和容器的数量（包括总计、失败、挂起、正在运行和已完成的数量）。
```

yuchaoran2011 commented on i18n/zh-cn/docusaurus-plugin-content-docs/version-0.12.1/api/cluster.md:
```diff
@@ -0,0 +1,128 @@
+
+## 集群利用率
+
+返回与群集资源利用率相关的统计数据
```
```suggestion
返回与集群资源利用率相关的统计数据
```

yuchaoran2011 commented on i18n/zh-cn/docusaurus-plugin-content-docs/version-0.12.1/api/scheduler.md:
```diff
@@ -0,0 +1,1318 @@
+
+**样例内容**
+
+For the default queue hierarchy (only `root.default` leaf queue exists) a similar response to the following is sent back to the client:
```
You meant to remove this line because it has been translated below?

yuchaoran2011 commented on i18n/zh-cn/docusaurus-plugin-content-docs/version-0.12.1/design/architecture.md:
```diff
@@ -0,0 +1,56 @@
+-->
+
+Apache YuniKorn (Incubating) 是一个用于容器编排器系统的轻量级通用资源调度器。
+它的创建是为了在大规模、多租户和云本地环境中高效地实现各种工作负载的细粒度资源共享。
```
```suggestion
它的创建是为了在大规模、多租户和云原生环境中高效地实现各种工作负载的细粒度资源共享。
```

yuchaoran2011 commented on i18n/zh-cn/docusaurus-plugin-content-docs/version-0.12.1/design/cross_queue_preemption.md:
```diff
@@ -0,0 +1,129 @@
+
+**以下是最好的事情:**
+
+- 抢占很快（多亏了“猎枪”），回收数千个 contaier 并只需要1秒。
```
```suggestion
- 抢占很快（多亏了“猎枪”），回收数千个容器只需要1秒。
```

yuchaoran2011 commented on i18n/zh-cn/docusaurus-plugin-content-docs/version-0.12.1/design/cross_queue_preemption.md:
```diff
@@ -0,0 +1,129 @@
+我们可以通过抢占来进行资源的分配。
+这并不意味着容器将在发出抢占权后立即停止。
+相反，RM可以控制向容器发送信号和终止容器之间的延迟。
+比如在K8s中终止POD：https://kubernetes.io/docs/concepts/workloads/pods/pod/#termination-of-pods
```
```suggestion
比如在K8s中终止Pod：https://kubernetes.io/docs/concepts/workloads/pods/pod/#termination-of-pods
```

yuchaoran2011 commented on i18n/zh-cn/docusaurus-plugin-content-docs/version-0.12.1/design/gang_scheduling.md:
```diff
@@ -0,0 +1,608 @@
+本设计将 _不_ 讨论核心端的广义抢占。
+
+## 通用流程
+该流由提交的 pod 触发，该 pod 触发应用程序创建。
```
```suggestion
该流程由提交的 pod 触发。第一个 pod 触发应用程序创建。
```

yuchaoran2011 commented on i18n/zh-cn/docusaurus-plugin-content-docs/version-0.12.1/design/gang_scheduling.md:
```diff
@@ -0,0 +1,608 @@
+必须将与帮派相关的字段添加到这两个消息中。
+
+分配释放请求和响应请求需要支持双向通信，并且需要进行重大更改。
+
```
I assume the English content below is yet to be translated?

cdmikechen replied:
@yuchaoran2011 
是的，这里有两个原因。
第一个是我在翻译 gang scheduling 的时候遇到了一些困难。在我翻译的过程中发现这篇文章里面存在着很多我难以理解的句子，我结合上下文也很难翻译和猜测出来原作者的意思；同时，这里面有一些单词拼接的专有词汇我还是有一些拿捏不准，不知道是完全翻译成中文还是保留原样，因为可能这些单词在实际的代码或者使用过程中可能会有一些语义在里面，翻译成中文后反倒会让读了文章的使用者对应不上关系。
第二个是我原来的翻译顺序是按照我自己文档目录树的顺序翻译的，但是考虑到读者的阅读顺序，从website开始按照官网的顺序进行翻译可能更加好。所以我将当时翻译一半的内容先停了下来。
我个人的想法是在后续的会议中如果我参与，可能会请教一下各位前辈一些专有名词的翻译。其实有很多的专有名词已经约定俗成，在很多情况下不翻译可能让读者更好地进行理解。

当前的翻译并不是全部的，为了保证文档的完整性，我暂时将未完成的部分也追加到目录下，以防止发生文档找不到的情况

yuchaoran2011 replied:
Thanks for the clarification. I agree it's hard to find counterparts in Chinese for some terminologies. You are welcome to join the [weekly community meeting](https://yunikorn.apache.org/community/get_involved) for Mandarin speakers to discuss with people. It's at 1pm China Standard Time on Wednesday.

yangwwei replied:
hi @cdmikechen thank you so much for initiating the effort!! and thanks for @yuchaoran2011 's help to review.
I do not think we need to wait for everything ready before committing patches, we can change YUNIKORN-1029 to an umbrella and do this incrementally. Actually, I am gonna propose in the next community, we should do a split for the reviewers for people who can help to review. 

for your questions
> 第一个是我在翻译 gang scheduling 的时候遇到了一些困难
Understood, that's why we need to work together more closely : )

> 第二个是我原来的翻译顺序是按照我自己文档目录树的顺序翻译的，但是考虑到读者的阅读顺序，从website开始按照官网的顺序进行翻译可能更加好。所以我将当时翻译一半的内容先停了下来。
I think we should focus on the user-facing doc first. such as quick start, download, roadmap, community, user guide etc. for developer-facing docs, that can come later, such as design docs.

chenya-zhang replied:
Thanks @cdmikechen for this great contribution!

+1 if we could split this PR into smaller ones. It will be much easier for more reviewers to help on the review process. :)

yuchaoran2011 reviewed (changes requested):
Thanks for initiating this great effort. A lot of work has gone into this PR. I haven't gone through everything but pointed out some typos and nits to be addressed for now

cdmikechen commented:
> Thanks for initiating this great effort. A lot of work has gone into this PR. I haven't gone through everything but pointed out some typos and nits to be addressed for now
@yuchaoran2011 
谢谢您的指导，您提出的一些意见我已经进行了更正，我在这里先以中文方式回复您。
由于我对yunikorn项目也暂时只在了解阶段，所以当前的文档中的确是会有很多的疏漏和不足。有一些词汇和语句我还是很不理解，所以会出现部分机翻的痕迹（一部分是有些部分的章节比较简单，我就直接使用翻译软件进行了翻译，然后简单校对一下；一部分是的确是我自己不理解一些语句，利用翻译软件翻译后再自己稍微润色一下尽量保证文档的连贯性和一定的可读性）。
目前的PR可能更接近粗翻，没有经过太精细的校对。如果您有时间，我想我可以先把一些我的疑问整理一下再请教您。

现在有2个方向：接下来先完成其他部分章节，然后再进行细翻；或者先保证当前文档的质量，再进行接下来的部分。您觉得哪个方向更好一点？

yuchaoran2011 commented:
I'm leaning towards the first option. It will be confusing for users looking for Chinese documentation to run into large paragraphs of English. Let's also hear thoughts from @yangwwei @chenya-zhang @HuangTing-Yao @0yukali0

yangwwei commented on i18n/zh-cn/docusaurus-plugin-content-pages/community/events.md:
```diff
@@ -0,0 +1,61 @@
+
+_Abstract_: Apache YuniKorn (Incubating) 今年早些时候发布了 v0.11 版本，其中包含了许多新功能和改进，比如 Gang 调度、REST API 增强和 Kubernetes 1.19 支持。
+在一个月内，我们计划发布主要的 v1.0.0 版本，它将支持 Kubernetes 1.20 和 1.21、改进的节点排序以及大量的小修复和增强功能！
+在本次会议中，我们将深入探讨在 Kubernetes 上使用临时占位 pod 背后的帮派调度实现、通过简化的调度器核心代码和新的节点排序算法显著提高性能。
```
感觉gang scheduling翻译成帮派调度有点奇怪
要不然这个就不要翻译，直接gang scheduling以避免误导用户

yangwwei commented on i18n/zh-cn/docusaurus-plugin-content-pages/community/get_involved.md:
```diff
@@ -0,0 +1,125 @@
+在成为 PPMC 成员之前，贡献者必须先成为提交者。
+提交者何时有资格被添加到 PPMC 没有严格的规定。
+管理委员会根据每个提交者的参与和影响做出决定。
+一般来说，对项目做出一致的代码或非代码贡献的提交者应被视为 PPMC 候选人。
```
> 一般来说，对项目做出一致的代码或非代码贡献的提交者应被视为 PPMC 候选人。

一般来说，对项目做出持续的代码或非代码贡献的贡献者应被视为 PPMC 候选人。

yangwwei commented on i18n/zh-cn/docusaurus-plugin-content-pages/community/how_to_contribute.md:
```diff
@@ -0,0 +1,107 @@
+如果您发现任何安全问题，请将漏洞报告发送至 security@apache.org，YuniKorn 安全团队将立即评估问题并与报告者一起制定修复计划。
+在与安全团队合作之前，请不要将问题透露给任何公共论坛。
+
+## 如果扔有问题？
```
如果仍有问题？

错别字扔

yangwwei commented on i18n/zh-cn/docusaurus-plugin-content-pages/community/roadmap.md:
```diff
@@ -0,0 +1,57 @@
+- [YUNIKORN-978](https://issues.apache.org/jira/browse/YUNIKORN-978) 准入控制器部署改进
+- [YUNIKORN-984](https://issues.apache.org/jira/browse/YUNIKORN-984) 实施用户和组限制（阶段 1）
+- [YUNIKORN-954](https://issues.apache.org/jira/browse/YUNIKORN-954) 移除旧的 REST api
+- [YUNIKORN-720](https://issues.apache.org/jira/browse/YUNIKORN-720) 改进了队列指标
```
对调度器资源队列指标监测系统的改进

yangwwei commented on i18n/zh-cn/docusaurus-plugin-content-pages/release-announce/0.10.0.md:
```diff
@@ -0,0 +1,54 @@
+# 发布公告 v0.10.0
+我们非常高兴地宣布 Apache YuniKorn (Incubating) 社区已投票发布 0.10.0。Apache YuniKorn (Incubating) 是一个独立的资源调度器，旨在管理和调度容器编排框架上的大数据工作负载，支持本地和云端的用例。
+
+此版本中添加的显著功能是联邦调度，YuniKorn 现在可以在 Kubernetes 上运行应用程序时提供联邦调度功能。除此之外，此版本还包括各种错误修复和改进。
```
此版本中添加的显著功能是 gang scheduling， YuniKorn 现在可以在 Kubernetes 上运行应用程序时提供gang scheduling 调度功能。除此之外，此版本还包括各种错误修复和改进。

yangwwei commented on i18n/zh-cn/docusaurus-plugin-content-pages/release-announce/0.10.0.md:
```diff
@@ -0,0 +1,54 @@
+从此版本开始，支持的 Kubernetes 版本已更新为 1.16.x、1.17.x 和 1.18.x。（早期版本支持 1.13.x、1.14.x 和 1.15.x）YuniKorn 支持矩阵主要支持 3 个主要的 Kubernetes 版本。
+
+### 联邦调度
+在这个版本中，YuniKorn 开始支持联邦调度。用户可以将联邦应用于需要联邦调度语义的应用，如 Spark、Tensorflow、Pytorch 等。YuniKorn 主动为联邦调度应用预留资源，它与 cluster-autoscaler 配合使用更高效。初始的支持已经在 Spark 上进行了很好的测试，它可以与 K8s 上的原生 Spark 或 Spark K8s operator 一起使用。有关如何启用和使用联邦调度的更多信息，请阅读 [此处](/docs/user_guide/gang_scheduling) 的文档。
```
请将 联邦调度 改名为 gang scheduling

yangwwei commented on i18n/zh-cn/docusaurus-plugin-content-pages/release-announce/0.11.0.md:
```diff
@@ -0,0 +1,57 @@
+-->
+# 发布公告 v0.11.0
+我们非常高兴地宣布 Apache YuniKorn (Incubating) 社区已投票发布 0.11.0。Apache YuniKorn (Incubating) 是一个独立的资源调度器，旨在管理和调度容器编排框架（如 Kubernetes）上的大数据工作负载，支持本地和云端的用例。
+此版本有几个 REST API、UI 可用性和联邦调度的改进。
```
联邦调度 - gang scheduling

yangwwei commented:
I have done the first-round review, I have skipped design docs in this round because I want to focus more on the user-facing docs. Please check my comments, besides that, a couple of more questions

1. Are we planning to translate the home page as well? 
2. I noticed that all the files are under `docusaurus-plugin-content-docs/version-0.12.1/`, is this only done for the v0.12.1 doc? I think we need to get the master doc done first. Note, our source docs are under https://github.com/apache/incubator-yunikorn-site/tree/master/docs, and version docs are all generated. I am not familiar with how docusaurus support different languages, but we need to make sure the CN docs are also generated from the master doc. 
3. Have you tested this locally and how it looks like? If you haven't done this, you can see how to do this locally in this here: https://github.com/apache/incubator-yunikorn-site#make-changes.

and a few suggestions

1. We need to use consistent terminology in the doc, such as gang scheduling, placement rules, resource quota, leaf queue, and so on. Once all review feedback is received, please don't forget to do a double-check on these terminology words.
2. Please focus on the user-facing doc first, that's gonna hit more users : )

cdmikechen commented:
@yangwwei 
Thank you for your review~ I will change it later. Let me answer these questions first.

> 1. Are we planning to translate the home page as well?

Yes, `docusaurus-plugin-content-pages` is yunikorn home website pages and the preliminary translation has been completed. 

> 2\. I noticed that all the files are under `docusaurus-plugin-content-docs/version-0.12.1/`, is this only done for the v0.12.1 doc? I think we need to get the master doc done first. Note, our source docs are under https://github.com/apache/incubator-yunikorn-site/tree/master/docs, and version docs are all generated. I am not familiar with how docusaurus support different languages, but we need to make sure the CN docs are also generated from the master doc.

I added these files completely according to [docusaurus official 2.0 document](https://docusaurus.io/zh-CN/docs/i18n/introduction).
And I will move these documents to current, which is the master version in the official website.

> 3\. Have you tested this locally and how it looks like? If you haven't done this, you can see how to do this locally in this here: https://github.com/apache/incubator-yunikorn-site#make-changes.

I have deployed and verified locally. In the next time, when the translation is relatively perfect, I can provide a docker image or build a private service to preview the new page.

> 1. We need to use consistent terminology in the doc, such as gang scheduling, placement rules, resource quota, leaf queue, and so on. Once all review feedback is received, please don't forget to do a double-check on these terminology words.

In addition to these listed words, I'll try to sort out a proper noun document this week. I will send it to slack `#dev` channel. So that you or other PPMC can review and see which words can be translated and which need to keep the original.

cdmikechen commented on i18n/zh-cn/docusaurus-plugin-content-docs/version-0.12.1/get_started/core_features.md:
```diff
@@ -0,0 +1,72 @@
+
+应用可以在工作队列中正确排队，排序策略决定哪个应用程序可以首先获得资源。
+这个策略可以是多种多样的，例如简单的 `FIFO`、`Fair`、`StateAware`或基于`Priority`的策略。
+队列可以维持应用的顺序，调度器根据不同的策略为作业分配相应的资源。这种行为更容易被理解和控制。
```
@yangwwei 
Queues can maintain the order of applications, and based on different policies, the scheduler allocates resources to jobs accordingly. The behavior is much more predictable.
这个是最后一句的原文，请问下是不是指定错了行？

cdmikechen commented on i18n/zh-cn/docusaurus-plugin-content-docs/version-0.12.1/get_started/core_features.md:
```diff
@@ -0,0 +1,72 @@
+在多租户环境中，许多用户共享集群资源。
+为了避免租户争夺资源或者可能的资源不足，需要做到更细粒度的公平性需求，以此来实现跨用户以及跨团队/组织的公平性。
+考虑到权重或优先级，一些更重要的应用可以获得超过其配额的更多的需求资源。
+这往往与资源预算有关，更细粒度的公平模式可以进一步提高资源控制。
```
@yangwwei 
This is often associated with resource budget, a more fine-grained fairness mode can further improve the expense control.
还有这一句，需要将 expense control 意译一下吗？

cdmikechen commented on i18n/zh-cn/docusaurus-plugin-content-docs/version-0.12.1/get_started/core_features.md:
```diff
@@ -0,0 +1,72 @@
+## 资源预留
+
+YuniKorn会自动为未完成的请求进行资源预订。
+如果Pod无法分配，YuniKorn将尝试在一个满足条件的节点上保留它，并在这个保留节点上暂时分配pod（在尝试其他节点之前）。
+这种机制可以避免这个Pod被后来提交的更小、更不挑剔的豆荚所挤占。
+此功能在批处理工作负载场景中非常重要，因为当对集群提交大量异构Pod时，很有可能一些Pod会被抛弃，即使它们提交得更早。
```
@yangwwei 
If a pod could not be allocated, YuniKorn will try to reserve it on a qualified node and tentatively allocate the pod on this reserved node (before trying rest of nodes). 
这个是第一句，我想请问下这句扩充了很多信息，是否需要回馈到原文进行修改？

yangwwei commented:
Sounds great, thank you for answering my questions. 

> I added these files completely according to docusaurus official 2.0 document.
And I will move these documents to current, which is the master version in the official website.

yes, please. Besides that, could you please check our doc here: https://github.com/apache/incubator-yunikorn-site#releasing-a-new-version about how to release a new version of doc. With the translated docs in this PR, what will these steps look like? Could you please elaborate, thanks!

cdmikechen commented:
@yuchaoran2011 @yangwwei @chenya-zhang 
Hi~ I’ve created some sub tasks under YUNIKORN-1029.

![image](https://user-images.githubusercontent.com/12069428/150039660-e927c95f-3aaf-4601-b270-bb60fd565643.png)

And I will focus on [YUNIKORN-1031 Add Chinese translation for homepage](https://issues.apache.org/jira/browse/YUNIKORN-1031) right now. I will follow the update of GitHub at any time and synchronize the new commit to this PR as soon as possible

cdmikechen commented:
@yuchaoran2011 @yangwwei @chenya-zhang
good news, I have deploy Chinese website in here <https://cdmikechen.github.io/incubator-yunikorn-site/zh-cn>
We can continue the discussion according to this page to see if there is any subsequent adjustment.

yangwwei commented:
hi @cdmikechen any updates?
Please split this PR into smaller ones and link them to the sub-tasks under https://issues.apache.org/jira/browse/YUNIKORN-1029. Thanks

cdmikechen commented:
@yangwwei 
Hi~ I have deleted the doc related pages and I will remind you after I merge your pr.

cdmikechen commented:
@yangwwei
Hi ~ I've merge codes to newest and translate why contents.

yangwwei commented:
hi @cdmikechen I thought you will split your PRs into smaller ones, that will be easier for review. Can you help to do that? 
For example, the first change I expect will be a PR for https://issues.apache.org/jira/browse/YUNIKORN-1031. Which only contains the necessary changes for introducing the CN language option and the translation for the home page.

Then you can create separate PRs for each of the sub-tasks, and we can assign different reviewers to review your changes more easily. Does that make sense?

cdmikechen commented:
@yangwwei 
抱歉，我现在可能理解了您的意思。我原来的考虑是把首页和如下的几个章节作为初次PR提交的主页内容。
![image](https://user-images.githubusercontent.com/12069428/151167239-03d17d51-05f3-48b8-ba0d-b8bbbf84ad13.png)
因为我会假设一个正常访问页面的朋友，一部分人是为了下载最新/指定版本软件，一部分人或者是为了查看一下release note，还有的人是为了查看文档寻求一些详细内容的解答。抛开doc部分内容来说，对于前一部分的人来讲，从一个中文页面转到英文界面会让人感觉有一定的阻断性。我当时觉得把 `docusaurus-plugin-content-pages` 内部分作为一次性的提交是可以的，所以在上一次提交的时候把其他doc章节的内容都进行了删除。
另外一点考量是一次性提交一个部分/章节内的几篇翻译文档应该是一个较为有效率的事情。我实际上在拆分子任务的时候，大概在心里计划了每次提交需要涉及多少部分。

不过，如果您认为我们的第一次PR应该只包含独立的页面，我会进行修改的。

yangwwei commented:
> @yangwwei 抱歉，我现在可能理解了您的意思。我原来的考虑是把首页和如下的几个章节作为初次PR提交的主页内容。 ![image](https://user-images.githubusercontent.com/12069428/151167239-03d17d51-05f3-48b8-ba0d-b8bbbf84ad13.png) 因为我会假设一个正常访问页面的朋友，一部分人是为了下载最新/指定版本软件，一部分人或者是为了查看一下release note，还有的人是为了查看文档寻求一些详细内容的解答。抛开doc部分内容来说，对于前一部分的人来讲，从一个中文页面转到英文界面会让人感觉有一定的阻断性。我当时觉得把 `docusaurus-plugin-content-pages` 内部分作为一次性的提交是可以的，所以在上一次提交的时候把其他doc章节的内容都进行了删除。 另外一点考量是一次性提交一个部分/章节内的几篇翻译文档应该是一个较为有效率的事情。我实际上在拆分子任务的时候，大概在心里计划了每次提交需要涉及多少部分。
> 
> 不过，如果您认为我们的第一次PR应该只包含独立的页面，我会进行修改的。

OK，谢谢你的解释，这样也是OK的。我会来review这一个。感谢！

yangwwei commented on i18n/zh-cn/docusaurus-plugin-content-pages/community/how_to_contribute.md:
```diff
@@ -0,0 +1,107 @@
+有人可能已经在处理它并且没有分配 ticket。
+具有拉取请求链接的 JIRA 将具有标签 `pull-request-available`，并且可以在问题链接下找到拉取请求的链接。
+
+对于任何不只是微不足道的更改，例如拼写错误或一行代码更改，讨论您在该问题上的预期方法是一个好主意。
```
> 讨论您在该问题上的预期方法是一个好主意

最好修改为： 在公开的渠道如slack或者邮件列表讨论您将如何解决这个问题将很有帮助。

yangwwei commented on i18n/zh-cn/docusaurus-plugin-content-pages/community/how_to_contribute.md:
```diff
@@ -0,0 +1,107 @@
+如果您不确定如何为特定组件编写测试，请在 JIRA 上寻求指导。
+
+请让您的补丁只修改了 JIRA 描述的问题。
+如果我们对每个补丁的范围保持记录，这对每个人都会更好。
```
> 请让您的补丁只修改了 JIRA 描述的问题。如果我们对每个补丁的范围保持记录，这对每个人都会更好。

您的补丁应该只包含对 JIRA 描述的问题的修复，超过改范围应该由新的 JIRA 来跟踪。这样的原则有利于代码的维护和审核。

yangwwei commented on i18n/zh-cn/docusaurus-plugin-content-pages/index.js:
```diff
@@ -0,0 +1,169 @@
+    description: (
+      <>
+      支持本地和云端用例。 
+      在云上运行时，它与自动缩放共同协作，以带来最大的资源弹性和更好的吞吐量。
```
自动缩放  -> 弹性伸缩控制器

yangwwei reviewed (changes requested):
Mostly look good, a few minor places need to be fixed.
Once they are fixed, I'll get this merged. Thank you!!

cdmikechen commented:
@yangwwei 
I've changed by your reviews. And I found we have added a new page (people.md) and I also add this to this PR.
The newest preview in <https://cdmikechen.github.io/incubator-yunikorn-site/zh-cn> , you can take a look at the final results.

yangwwei reviewed (approved):
Looks all good now, merging to the master branch.
Thanks a lot for getting this done! @cdmikechen
